packages/transaction-controller/src/utils/resimulate.ts

```diff
@@ -0,0 +1,288 @@
+import type { Hex } from '@metamask/utils';
+import { createModuleLogger, remove0x } from '@metamask/utils';
+import { BN } from 'bn.js';
+import { isEqual } from 'lodash';
+
+import { projectLogger } from '../logger';
+import type {
+  SimulationBalanceChange,
+  SimulationData,
+  TransactionMeta,
+  TransactionParams,
+} from '../types';
+import { getPercentageChange } from './utils';
+
+const log = createModuleLogger(projectLogger, 'resimulate');
+
+export const RESIMULATE_PARAMS = ['to', 'value', 'data'] as const;
+export const BLOCKAID_RESULT_TYPE_MALICIOUS = 'Malicious';
+export const VALUE_COMPARISON_PERCENT_THRESHOLD = 5;
+export const BLOCK_TIME_ADDITIONAL_SECONDS = 60;
+
+export type ResimulateResponse = {
+  blockTime?: number;
+  resimulate: boolean;
+};
+
+/**
+ * Determine if a transaction should be resimulated.
+ * @param originalTransactionMeta - The original transaction metadata.
+ * @param newTransactionMeta - The new transaction metadata.
+ * @returns Whether the transaction should be resimulated.
+ */
+export function shouldResimulate(
```

Comment:
[suggestion] 

related to the [comment below](https://github.com/MetaMask/core/pull/4792/files#r1819055361), I think prepending the function name with a verb in addition to the description
```
@returns Whether the transaction should be resimulated.
```

implies this method may return a boolean. However, it seems to return the object
```
  return {
    blockTime,
    resimulate,
  };
```

might i suggest renaming the function name to possibly `generateResimulateObject`

Reply:
I think the current name adds more conceptual clarity, though I'll update the JSDoc since it's out of date.